src/frontend/js/toggle-dashboard-section.ts

```diff
@@ -18,3 +20,97 @@ async function toggleDashboardSection({ target }: Event): Promise<void> {
   content.classList.toggle("max-h-160");
   content.classList.toggle("max-h-0");
 }
+
+
+async function loadRssFeed() {
+  const feedWidget = document.querySelector(".rss-feed");
+
+  if (feedWidget) {
+    const feedLoading = document.getElementById("feed-loading");
+    const reloadButton = document.getElementById("reload");
+
+    const handleError = () => {
+      feedLoading.classList.add("hidden");
+      reloadButton.classList.remove("hidden");
+      reloadButton.addEventListener("click", loadRssFeed);
+    };
+    
+    feedLoading.classList.remove("hidden");
+    reloadButton.classList.add("hidden");
+    const url = feedWidget.attributes[1].value;
+
+    // set timeout
+    const controller = new AbortController();
+    const timeoutId = setTimeout(() => {
+      controller.abort();
+      handleError();
+    }, 5000);
+
+    fetch(url, {signal: controller.signal}).then(async (response: Response) => {
+      if (response.status === 200 && response.ok) {
+        feedLoading.classList.add("hidden");
+        clearTimeout(timeoutId);
+        let parser = new DOMParser;
+        let feedHtml = parser.parseFromString(
+          parser.parseFromString(await response.text(), "text/xml").documentElement.innerHTML, "text/html");
```

Comment:
```suggestion
       let feedHtml = parser.parseFromString(rawFeed, "text/xml");
```
Just calling the inner `parseFromString` should be enough to get the HTML document